.github/workflows/update-deps.yml

```diff
@@ -0,0 +1,71 @@
+name: Update NPM dependencies
+
+on:
+  schedule:
+  - cron: "0 5 * * 1"
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        deps: [production, dev-only]
+
+    steps:
+      - name: Fetch changes
+        # https://github.com/marketplace/actions/checkout
+        uses: actions/checkout@v2
+        with:
+          ref: master
```

Comment:
There is [GITHUB\_REF](https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables) which points to branch which triggered workflow. Since scheduled workflows are only triggered by default branch, it should work fine. We can also allow possibility to overwrite this via env variables.